tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_rand_uint.h

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "ckernel_sfpu_converter.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE>
+inline void rand_uint_init(uint seed) {
+    init_prng_seed(seed);
+}
+
+template <bool APPROXIMATION_MODE>
+inline void rand_uint() {
+#pragma GCC unroll 0
+    for (int d = 0; d < 8; d++) {
+        TTI_SFPMOV(0, 9, p_sfpu::LREG0, 8);
+        TTI_SFPSETSGN(0, p_sfpu::LREG0, p_sfpu::LREG0, 1);
+        TTI_SFPSTORE(0, 4, 3, 0);
+        TTI_SFPIADD(0, p_sfpu::LREG0, p_sfpu::LREG0, 5);
```

Comment:
@BuiChiTrung Can you explain in the above how the `SFPSTORE` function is used before `SFPIADD` and it is giving the correct results? The `SFPSTORE` function is the one that writes the lreg (in the case above it is LREG0) to dst. So I am confused on why the SFPIADD is called after the result is written back to dst?

Reply:
or is the LREG0 above maintaining value like a counter?

Reply:
You're right. `LREG0` is used as a counter, I removed the counter init code in this [commit](https://github.com/tenstorrent/tt-metal/pull/13735/commits/6837a87e6b0fdf175f7994dcaf5741805f494b99). It's my bad not removing `SFPIADD` here. Just removed it.